src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -11,6 +11,32 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public int hashCode() {
+        return 17 * 31 + (width == null ? 0 : width.hashCode())
+                * 17 * 31 + (length == null ? 0 : length.hashCode())
+                * 17 * 31 + (color == null ? 0 : color.hashCode());
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || obj.getClass() != getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) obj;
+        if ((this.color == null && rectangle.color == null)
+                && (this.length == null && rectangle.length == null)
+                && (this.width == null && rectangle.width == null)) {
+            return true;
+        }
+        return this.width != null && this.width.equals(rectangle.width)
+                && this.length != null && this.length.equals(rectangle.length)
+                && this.color != null && this.color.equals(rectangle.color);
```

Comment:
let's not use `this` keyword - it is redundant 

also, lets combine lines 30-37 in a single statement
if both color and rectangle.color are null, the following statement will be true

`color == rectangle.color`